### PR TITLE
Update pandas to 2.1.1

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- Update to 2.1.1

https://pandas.pydata.org/pandas-docs/version/2.1.1/getting_started/install.html
https://github.com/pandas-dev/pandas/blob/v2.1.1/pyproject.toml

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+if [ "$(uname -m)" = "ppc64le" ]; then
+  export PYTHONUNBUFFERED=1
+fi
+
+mkdir builddir
+$PYTHON -m build --wheel --no-isolation --skip-dependency-check \
+    -Cbuilddir=builddir \
+    || (cat builddir/meson-logs/meson-log.txt && exit 1)
+$PYTHON -m pip install dist/pandas*.whl
```
Comment (skupr-anaconda) [inline recipe/build.sh:15]: Could you point me, please, where this code come from? I didn't find it on github or docs, even conda-forge still uses `pip install .`

**Review by skupr-anaconda — APPROVED** [review]
LGTM but the build scripts

**Comment by skupr-anaconda** [comment]
`ld[73579]: Library not loaded: @rpath/libclang-cpp.14.dylib`

@cbouss If you reopen PR the issue will gone. But I would ask the Sirius team why it happens periodically.

File: recipe/build.sh (hunk 1 above)
Comment (cbouss) [reply recipe/build.sh:15]: When using pip install, meson-log.txt ends up in a temporary directory.
By using python-build, we can display the content of meson-log.txt on error.
scipy and numpy feedstocks have similar code in their build script.

Comment (JeanChristopheMorinPerso) [reply recipe/build.sh:15]: We can pass build options to pip too. But it's fine if we python-build.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,42 +1,35 @@
...
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
   host:
     - python
-    - pip
-    - setuptools >=61.0.0
+    - python-build
+    - meson-python ==0.13.1
+    - meson ==1.2.1
     - wheel
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:27]:
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -45,96 +38,98 @@ requirements:
     - pytz >=2020.1
     - python-tzdata >=2022.1
     # Recommended dependencies to achieve large speedups
-    # see https://pandas.pydata.org/pandas-docs/version/2.0.3/getting_started/install.html#recommended-dependencies
+    # see https://pandas.pydata.org/pandas-docs/version/2.1.1/getting_started/install.html#performance-dependencies-recommended
     # NOTE: For numexpr, there is disagreement between pyproject.toml and internal documentation.  We went with the higher of the two.
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:42]: This comment can be removed.

**Pull request merged**